packages/http-server/src/server.ts

```diff
@@ -1,28 +1,73 @@
-const express = require('express');
+import { IFilesystemLoaderOpts } from '@stoplight/prism-core';
 import { createInstance } from '@stoplight/prism-http';
+import { IHttpMethod, TPrismHttpInstance } from '@stoplight/prism-http/types';
+import * as fastify from 'fastify';
+import { IncomingMessage, Server, ServerResponse } from 'http';
+
 import { getHttpConfigFromRequest } from './getHttpConfigFromRequest';
+import { IPrismHttpServer, IPrismHttpServerOpts } from './types';
+
+export const createServer = <LoaderInput = IFilesystemLoaderOpts>(
+  loaderInput: LoaderInput,
+  opts: IPrismHttpServerOpts<LoaderInput> = {}
+): IPrismHttpServer<LoaderInput> => {
+  const server = fastify<Server, IncomingMessage, ServerResponse>();
+
+  const prism = createInstance<LoaderInput>({
+    config: getHttpConfigFromRequest,
+    ...(opts.components || {}),
+  })(loaderInput);
+
+  server.all('*', {}, replyHandler(prism));
+
+  const prismServer: IPrismHttpServer<LoaderInput> = {
+    get prism() {
+      return prism;
+    },
+
+    get fastify() {
+      return server;
+    },
+
+    listen: async (...args) => {
+      return server.listen(...args);
+    },
+  };
+
+  return prismServer;
+};
+
+const replyHandler = (
+  prism: TPrismHttpInstance
+): fastify.RequestHandler<IncomingMessage, ServerResponse> => {
+  return async (request, reply) => {
+    const { req } = request;
+
+    try {
+      const response = await prism.process({
+        method: (req.method || 'get') as IHttpMethod,
+        url: {
+          path: req.url || '/',
+          query: request.query,
```

Comment:
One remark. `baseUrl` is optional here but if not provided `prism` will assume the path is relative (e.g. it will not take `servers` into account)

Reply:
Yeah, this is not really complete - see https://stoplightio.atlassian.net/browse/SL-216.

It is the reason that the test at https://github.com/stoplightio/prism/pull/161/files#diff-8dcd0f08ab8cb99f8132d693d8bf9682R72 is commented out.

Basically, pushing the URL parsing to the library user (http-server package in this case) is kind of annoying, and we might want to consider just accepting a simple URL string and handling the parsing internally in the http package. Going to close this one out for now, but feel free to respond in SL-216.